xbmc/FileItem.cpp

```diff
@@ -382,19 +381,10 @@ const CFileItem& CFileItem::operator=(const CFileItem& item)
     m_pvrChannelInfoTag = NULL;
   }
 
-  if (item.HasPVRRecordingInfoTag())
-  {
-    m_pvrRecordingInfoTag = GetPVRRecordingInfoTag();
-    if (m_pvrRecordingInfoTag)
-      *m_pvrRecordingInfoTag = *item.m_pvrRecordingInfoTag;
-  }
+  if (item.m_pvrRecordingInfoTag)
+    m_pvrRecordingInfoTag = item.m_pvrRecordingInfoTag;
```

Comment:
I think you are right, but the current implementation is buggy then, because it lets this' data untouched (even if it is not NULL) in case the other item has no recording info. other.recinfo = NULL => this.recinfo = (whatever-it-was-before, e.g. non-null). 

Do I have a brainfart or has the current implemenation really such a stupid bug?